### ISSUE TITLE
Required modifications to existing pretrained networks to allow CAM

### ISSUE THREAD

**Issue by ChrisParry** [metalbubble/CAM#4]
I have tried your demo using the pre-trained CAM models. I now want to try it with my own pre-trained Caffe CNNs. 

I would be very grateful if you could demonstrate the modifications that are required to convert a  pre-trained non-CAM model to the CAM format. Is it as simple as modifying the prototxt files directly? Thanks.

**Comment by metalbubble** [comment]
https://github.com/metalbubble/CAM/blob/master/models/deploy_vgg16CAM.prototxt#L324

You could refer to this prototxt file. Basically you need to remove all the fully connected layers in your original prototxt, then add a convolutional layer, average pooling layer (here you need to know the size of the feature map), and the final fc layer for softmax. Then you could fine-tune the modified prototxt file on your data for several epochs.

**Comment by kishensurajp** [comment]
In the finetuning step , do you completely freeze the original conv layers?

**Comment by mrgloom** [comment]
Why adding additional conv layer is needed? For example original VGG16 network have last conv layer with 512 planes, so I remove all layers after this last conv layer and then add new layers like `last_vgg16_conv->GAP->Dense`. And I even tried to finetune only dense layer looks like it work, at least for ImageNet-like classes.
